File: plugin/integration/scala-plugin/src/scala/org/intellij/plugin/zeppelin/scala/worksheet/extensions/WorksheetNotebookHolder.scala
package org.intellij.plugin.zeppelin.scala.worksheet.extensions

import com.intellij.psi.PsiFile
import org.intellij.plugin.zeppelin.scala.worksheet.settings.ZeppelinWorksheetFileSettings
import org.intellij.plugin.zeppelin.service.FileNotebookHolder

class WorksheetNotebookHolder extends FileNotebookHolder {
  override def contains(psiFile: PsiFile): Boolean = ZeppelinWorksheetFileSettings.hasNotebookId(psiFile)

  override def notebookId(psiFile: PsiFile): String = ZeppelinWorksheetFileSettings.getLinkedNotebookId(psiFile)
}

File: zeppelin/integration/scala-plugin/src/org/intellij/plugin/zeppelin/scala/worksheet/extensions/WorksheetNotebookHolder.kt
package org.intellij.plugin.zeppelin.scala.worksheet.extensions

import com.intellij.psi.PsiFile
import org.intellij.plugin.zeppelin.extensionpoints.FileNotebookHolder
import org.intellij.plugin.zeppelin.scala.worksheet.settings.ZeppelinFileSettings

class WorksheetNotebookHolder : FileNotebookHolder {
    override fun getNotebookId(psiFile: PsiFile): String =
            ZeppelinFileSettings.getLinkedNotebookId(psiFile)

    override fun contains(
            psiFile: PsiFile): Boolean = ZeppelinFileSettings.hasNotebookId(psiFile)
}